From df7f963c1fbe61ddec969e58c8b9a9f7663f1065 Mon Sep 17 00:00:00 2001
From: Avinash Sharma <avinash.sharma@cognizant.com>
Date: Mon, 3 Oct 2016 15:25:08 +0530
Subject: [PATCH] Fixed udhcpc command for Raspberry Pi platform.


--- a/source/scripts/init/c_registration/02_bridge.c
+++ b/source/scripts/init/c_registration/02_bridge.c
@@ -51,7 +51,7 @@
 const char* SERVICE_NAME            = "bridge";
 #ifdef INTEL_PUMA7
 const char* SERVICE_DEFAULT_HANDLER = "/etc/utopia/service.d/service_bridge_puma7.sh";
-#elif (_COSA_BCM_ARM_)
+#elif defined (_COSA_BCM_ARM_) && ! defined (_PLATFORM_RASPBERRYPI_)
 const char* SERVICE_DEFAULT_HANDLER = "/etc/utopia/service.d/service_bridge_tchxb6.sh";
 #else
 const char* SERVICE_DEFAULT_HANDLER = "/etc/utopia/service.d/service_bridge.sh";

diff --git a/source/scripts/init/c_registration/02_multinet.c b/source/scripts/init/c_registration/02_multinet.c
old mode 100644
new mode 100755
index 9b6b4a4..f2ceef3
--- a/source/scripts/init/c_registration/02_multinet.c
+++ b/source/scripts/init/c_registration/02_multinet.c
@@ -52,7 +52,7 @@ const char* SERVICE_NAME            = "multinet";
 
 #ifdef INTEL_PUMA7
 const char* SERVICE_DEFAULT_HANDLER = "/etc/utopia/service.d/vlan_util_xb6.sh";
-#elif (_COSA_BCM_ARM_)
+#elif defined (_COSA_BCM_ARM_) && ! defined (_PLATFORM_RASPBERRYPI_)
 const char* SERVICE_DEFAULT_HANDLER = "/etc/utopia/service.d/vlan_util_tchxb6.sh";
 #else
 const char* SERVICE_DEFAULT_HANDLER = "/etc/utopia/service.d/service_multinet_exec";
@@ -92,7 +92,7 @@ const char* SERVICE_CUSTOM_EVENTS[] = {
     "multinet-up|/etc/utopia/service.d/vlan_util_xb6.sh|"TUPLE_FLAG_EVENT,
     "lnf-setup|/etc/utopia/service.d/vlan_util_xb6.sh|"TUPLE_FLAG_EVENT, 
     NULL };
-#elif (_COSA_BCM_ARM_)
+#elif defined (_COSA_BCM_ARM_) && ! defined (_PLATFORM_RASPBERRYPI_)
 const char* SERVICE_CUSTOM_EVENTS[] = { 
     "multinet-syncNets|/etc/utopia/service.d/vlan_util_tchxb6.sh|NULL|"TUPLE_FLAG_EVENT,
     "multinet-syncMembers|/etc/utopia/service.d/vlan_util_tchxb6.sh|NULL|"TUPLE_FLAG_EVENT,
@@ -112,7 +112,7 @@ const char* SERVICE_CUSTOM_EVENTS[] = {
 
 void srv_register(void) {
    sm_register(SERVICE_NAME, SERVICE_DEFAULT_HANDLER, SERVICE_CUSTOM_EVENTS);
-#ifndef _COSA_BCM_ARM_ || ifndef INTEL_PUMA7
+#ifndef _COSA_BCM_ARM_ || ifndef INTEL_PUMA7 || ifdef _PLATFORM_RASPBERRYPI_
    system("/etc/utopia/service.d/service_multinet/handle_sw.sh initialize");
 #endif
 }
2.9.3

