diff --git a/source/service_wan/service_wan.c b/source/service_wan/service_wan.c
index 83c1937..9e8830e 100644
--- a/source/service_wan/service_wan.c
+++ b/source/service_wan/service_wan.c
@@ -535,7 +535,11 @@ static int wan_addr_set(struct serv_wan *sw)
             vsystem("((nfq_handler 4 %s &)&)", val);
             sysevent_set(sw->sefd, sw->setok, "parcon_nfq_status", "started", 0);
         }
+#if !defined(_PLATFORM_RASPBERRYPI_)
         vsystem("firewall && gw_lan_refresh && execute_dir /etc/utopia/post.d/ restart");
+#else
+        vsystem("firewall && execute_dir /etc/utopia/post.d/ restart");
+#endif
     } else if(bridgeMode != 0 && strcmp(lanstatus, "stopped") == 0 ) {
     	vsystem("firewall && execute_dir /etc/utopia/post.d/ restart");
     } else {
@@ -544,10 +548,15 @@ static int wan_addr_set(struct serv_wan *sw)
 	if(strcmp(mischandler_ready,"true") == 0)
 	{
 		//only for first time
+#if !defined(_PLATFORM_RASPBERRYPI_)
 		fprintf(stderr, "[%s] ready is set from misc handler. Doing gw_lan_refresh\n", PROG_NAME);
 		system("gw_lan_refresh ");
+#endif
 		sysevent_set(sw->sefd, sw->setok, "misc-ready-from-mischandler", "false", 0);
 	}
+#if defined(_PLATFORM_RASPBERRYPI_)
+	vsystem("execute_dir /etc/utopia/post.d/ restart");
+#endif
         fprintf(stderr, "[%s] start firewall fully\n", PROG_NAME);
         printf("%s Triggering RDKB_FIREWALL_RESTART\n",__FUNCTION__);
         sysevent_set(sw->sefd, sw->setok, "firewall-restart", NULL, 0);
